### PR TITLE
When saving magento product, reverb sync should be backgrounded [Fix #207]

### LINKED ISSUE #207: When saving magento product, reverb sync should be backgrounded
This seems strange - if reverb is slow/down it will affect the magento save of the product in the catalog. Is this how other plugins are done? I think we should be backgrounding this

cc @zztimur @dunagan5887 @StevenWolfe

### PR DESCRIPTION
**Pull request by dunagan5887**
Fix #207 Tested this on 1.7 and 1.9 and it looks good to go.  Tasks are executing via crontab execution as desired

### CODE REVIEW

File: app/code/community/Reverb/ReverbSync/Helper/Data.php [hunk 1]
```diff
@@ -33,6 +33,7 @@ public function createOrUpdateReverbListing($product, $do_not_allow_creation = f
     {
         // Create empty wrapper in the event an exception is thrown
         $listingWrapper = Mage::getModel('reverbSync/wrapper_listing');
+        /* @var $listingWrapper Reverb_ReverbSync_Model_Wrapper_Listing */
```
Comment (skwp) [inline app/code/community/Reverb/ReverbSync/Helper/Data.php]: commented out code?

File: app/code/community/Reverb/ReverbSync/Helper/Sync/Product.php [hunk 2]
```diff
@@ -131,8 +131,9 @@ public function executeSimpleProductSync($simpleProduct, $do_not_allow_creation
         if ($productType == Mage_Catalog_Model_Product_Type::TYPE_SIMPLE)
         {
             //pass the data to create or update the product in Reverb
-            $listingWrapper = Mage::helper('ReverbSync/data')
-                                ->createOrUpdateReverbListing($simpleProduct, $do_not_allow_creation);
+            $reverbSyncHelper = Mage::helper('ReverbSync/data');
+            /* @var $reverbSyncHelper Reverb_ReverbSync_Helper_Data */
```
Comment (skwp) [inline app/code/community/Reverb/ReverbSync/Helper/Sync/Product.php]: ?

File: app/code/community/Reverb/ReverbSync/Helper/Data.php (hunk 1 above)
Comment (dunagan5887) [inline app/code/community/Reverb/ReverbSync/Helper/Data.php]: /* @var $listingWrapper Reverb_ReverbSync_Model_Wrapper_Listing */

Comments such as this serve multiple purposes. One, it allows someone reviewing the code to know which classes the code intended to be using. Second, when using an IDE (such as PHPStorm which I use) comments such as these allow the IDE to verify that methods being called with object $listingWrapper not only exist, but are returning variables of the expected type (due to PHP comment docs). Third, if you can CTRL+click on "setMagentoProduct" in line 37 and the IDE will take you to the definition of the method.

These comments will help both in future development/refactoring of the model and for developers who are vetting out the code when deciding whether to include it on a Magento instance

File: app/code/community/Reverb/ReverbSync/Helper/Sync/Product.php (hunk 2 above)
Comment (dunagan5887) [inline app/code/community/Reverb/ReverbSync/Helper/Sync/Product.php]: Again, adding comments to denote the class of variable $reverbSyncHelper to make code traversal easier

File: app/code/community/Reverb/ReverbSync/Helper/Data.php (hunk 1 above)
Comment (skwp) [inline app/code/community/Reverb/ReverbSync/Helper/Data.php]: ok thanks

**Pull request merged**